src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,152 @@
+        System.out.println("    Noted. I've removed this task:");
+        System.out.println(list.get(index-1).getLabel());
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void handleMarkTask(ArrayList<Task> list, int indexToMark) {
+        list.get(indexToMark).setIsDone(true);
+        System.out.println("____________________________________________________________");
+        System.out.println("    Nice! I've marked this task as done:");
+        System.out.println(list.get(indexToMark).getLabel());
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void handleUnmarkTask(ArrayList<Task> list, int indexToUnmark) {
+        list.get(indexToUnmark).setIsDone(false);
+        System.out.println("____________________________________________________________");
+        System.out.println("    OK, I've marked this task as not done yet:");
+        System.out.println(list.get(indexToUnmark).getLabel());
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void handleExitProgram() {
+        System.out.println("____________________________________________________________");
+        System.out.println("    Bye. Hope to see you again soon!");
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void handleExceptions(Exception e) {
+        switch (e.getClass().getName()) {
+            case "duke.IllegalCommandException":
+                IllegalCommandException commandError = (IllegalCommandException) e;
+                if (commandError.type == IllegalCommandExceptionType.COMMAND_DOES_NOT_EXIST) {
+                    System.out.println("____________________________________________________________");
+                    System.out.println("    Error: The command '" + commandError.value + "' does not exist!");
+                    System.out.println("____________________________________________________________");
+                } else if (commandError.type == IllegalCommandExceptionType.MISSING_VALUE) {
+                    System.out.println("____________________________________________________________");
+                    System.out.println("    Error: Missing value after '" + commandError.value + "' command!");
+                    System.out.println("____________________________________________________________");
+                } else if (commandError.type == IllegalCommandExceptionType.MISSING_PARAMETER) {
+                    System.out.println("____________________________________________________________");
+                    System.out.println("    Error: This command requires the '" + commandError.value + "' parameter!");
+                    System.out.println("____________________________________________________________");
+                }
```

Comment:
Might want to consider refactoring the if-else structure out. Similarly for the if-else conditions under other exception.